com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/DebugAdapter.java

```diff
@@ -72,46 +68,25 @@ public DebugAdapter(BiConsumer<Events.DebugEvent, Boolean> consumer, IProviderCo
         try {
             if (debugContext.isVmTerminated()) {
                 // the operation is meaningless
-                return response;
+                return CompletableFuture.completedFuture(response);
             }
             List<IDebugRequestHandler> handlers = requestHandlers.get(command);
             if (handlers != null && !handlers.isEmpty()) {
                 for (IDebugRequestHandler handler : handlers) {
-                    handler.handle(command, cmdArgs, response, debugContext);
+                    CompletableFuture<Messages.Response> result = handler.handle(command, cmdArgs, response, debugContext);
```

Comment:
Suggest also guard the handle call in future like this:
return CompletableFuture.supplyAsync(() -> future.get())

So whatever happens in the handler won't affect the request loop. For example, we don't even need to check whether the result is null because it will automatically throw a NPE.

Reply:
Fixed.
`CompletableFuture.thenCompose` is more suitable for our scenario.